### PR TITLE
Improved performance on PreparedStatementReadConsumer with setNull(type)

### PR DESCRIPTION
**Pull request by nsvir**

### CODE REVIEW

File: src/main/java/io/github/anthorx/parquet/sql/write/NullTypeConsumer.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import org.apache.parquet.schema.Type;
+
+import java.sql.Types;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+/**
+ * create the Consumer that calls ReadRecordConsumer.setNull with the proper type
+ */
+public class NullTypeConsumer {
```
Comment (anthoRx) [inline src/main/java/io/github/anthorx/parquet/sql/write/NullTypeConsumer.java:15]: The class name lie, it's not a Consumer as we define it into the library (Java Consumer).
Find another name !

File: src/main/java/io/github/anthorx/parquet/sql/write/NullTypeConsumer.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  public static Consumer<ReadRecordConsumer> get(Type currentField) {
```
Comment (anthoRx) [inline src/main/java/io/github/anthorx/parquet/sql/write/NullTypeConsumer.java:17]: get what ? It's not clear.

File: src/main/java/io/github/anthorx/parquet/sql/write/JDBCWriter.java [hunk 3]
```diff
@@ -88,23 +87,22 @@ public void write() throws IOException, SQLException {
...
 
-  private RecordField createNullRecordField(Type currentField) {
-    RecordField<Object> recordField = new RecordField<>(currentField.getName(), null);
-    BiConsumer<ReadRecordConsumer, Object> objectRecordConsumer = ReadRecordConsumer::setObject;
-    recordField.addReadConsumer(objectRecordConsumer);
-
+  private RecordField<?> createNullRecordField(Type currentField) {
+    Consumer<ReadRecordConsumer> setNull = NullTypeConsumer.get(currentField);
+    RecordField<?> recordField = new RecordField<>(currentField.getName(), null);
+    recordField.addReadConsumer(((readRecordConsumer, o) -> setNull.accept(readRecordConsumer)));
```
Comment (anthoRx) [inline src/main/java/io/github/anthorx/parquet/sql/write/JDBCWriter.java:105]: I think this should be under the responsability of your class "NullTypeConsumer".
A function with this signature :
```java
<T> BiConsumer<ReadRecordConsumer, T> createNullReadRecordConsumer(Type currentField) {
```

**Review by anthoRx — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/anthorx/parquet/sql/read/PreparedStatementRecordConsumer.java [hunk 4]
```diff
@@ -180,8 +177,19 @@ public void setObject(Object value) {
...
+    try {
+      this.preparedStatement.setNull(getNextIndex(), typeId);
+    } catch (SQLException e) {
+      addError(e);
+    }
+  }
+
   @Override
   public void close() throws SQLException {
+    Connection connection = this.preparedStatement.getConnection();
```
[the remark this answers is not in the document]
Comment (nsvir) [reply src/main/java/io/github/anthorx/parquet/sql/read/PreparedStatementRecordConsumer.java:191]: I agree, parquet-sql should not manipulate a preparedStatement but only readRecordConsumer

File: src/main/java/io/github/anthorx/parquet/sql/write/JDBCWriter.java (hunk 3 above)
Comment (nsvir) [reply src/main/java/io/github/anthorx/parquet/sql/write/JDBCWriter.java:105]: agreed

**Pull request merged**